### PR TITLE
[SIMD.js] Map SSE1 partial load and store methods to custom intrinsics

### PR DESCRIPTION
**Pull request by huningxin**

### CODE REVIEW

**Comment by huningxin** [comment]
To address issue https://github.com/kripken/emscripten/issues/3119

**Comment by huningxin** [comment]
The LLVM part is in PR https://github.com/kripken/emscripten-fastcomp/pull/58

**Comment by huningxin** [comment]
I believe the better solution is to go [Masked Vector Load and Store Intrinsics](http://llvm.org/docs/LangRef.html#llvm-masked-load-intrinsics) when it lands in emscripten-fastcomp.

**Comment by huningxin** [comment]
@sunfishcode @juj , please take a look. Thanks!

File: system/include/emscripten/vector.h [hunk 1]
```diff
@@ -42,6 +42,11 @@ int32x4 emscripten_int32x4_select(int32x4 __a, int32x4 __b, int32x4 __c) __attri
 float32x4 emscripten_float32x4_fromInt32x4Bits(int32x4 __a) __attribute__((__nothrow__, __const__));
 float32x4 emscripten_float32x4_fromInt32x4(int32x4 __a) __attribute__((__nothrow__, __const__));
 
+float32x4 emscripten_float32x4_loadx(const void *__p) __attribute__((__nothrow__, __const__));
+float32x4 emscripten_float32x4_loadxy(const void *__p) __attribute__((__nothrow__, __const__));
+void emscripten_float32x4_storex(const void *__p, float32x4 __a) __attribute__((__nothrow__));
+void emscripten_float32x4_storexy(const void *__p, float32x4 __a) __attribute__((__nothrow__));
+
```
Comment (sunfishcode) [inline system/include/emscripten/vector.h]: Any reason not to add loadxyz and storexyz while we're here?

File: system/include/emscripten/xmmintrin.h [hunk 2]
```diff
@@ -53,17 +53,13 @@ _mm_load_ps(const float *__p)
 static __inline__ __m128 __attribute__((__always_inline__))
 _mm_loadl_pi(__m128 __a, const void /*__m64*/ *__p)
 {
-  // TODO: This actually corresponds to the SIMD.float32x4.loadXY function in
-  // SIMD.js. Use that instead.
-  return (__m128){ ((const float*)__p)[0], ((const float*)__p)[1], __a[2], __a[3] };
+  return __builtin_shufflevector(emscripten_float32x4_loadxy(__p), __a, 0, 1, 4, 5);
```
Comment (sunfishcode) [inline system/include/emscripten/xmmintrin.h]: This looks good. The TODO comment this replaces didn't anticipate the shuffle, but even so, this is better than what we had before.

**Comment by sunfishcode** [comment]
Looks good. I decided not to hold this up for loadxyz and storexyz; they can easily be added later.

**Comment by kripken** [comment]
This was merged to master by mistake a think. Please merge it to incoming and back it out of master (push -f).

**Comment by kripken** [comment]
I can do that now.

**Comment by kripken** [comment]
Done.

**Comment by huningxin** [comment]
@sunfishcode and @kripken , thanks for the review and merging! And I apologized that I made the PR to wrong branch.

**Comment by juj** [comment]
This looks good to me. However running the SSE1 test after this now fails. Not sure if the failure is caused by this PR, or if something else is related to the regression. See #3123.

**Pull request merged**